### ISSUE TITLE
Menu unusable on Pantheon DE

### ISSUE THREAD

**Issue by OPNA2608** [BespokeSynth/BespokeSynth#680]
Using version 1.1.0, the menu at the top is completely obscured by the title bar, and it looks like window content is missing at the bottom.

I'm using the [Pantheon desktop environment](https://elementary.io), on NixOS 21.05 and 21.11. I've observed this while testing our package bump in https://github.com/NixOS/nixpkgs/pull/144708.

![Bildschirmfoto_2021-12-08_12-17-08](https://user-images.githubusercontent.com/23431373/145199892-8eebbab5-413f-469c-9125-731117574f8f.png)

**Comment by KottV** [comment]
Can you test with some other JUCE App(s)?
No such problem in any other DE/WM.

**Comment by KottV** [comment]
also does change ui_scale in the settings something with that?

**Comment by OPNA2608** [comment]
> Can you test with some other JUCE App(s)?
> No such problem in any other DE/WM.

Could you point me to a JUCE app that I should test, ideally one [we've already packaged](https://search.nixos.org/packages)? [ADLplug](https://github.com/jpcima/ADLplug) is the only other one I know of and it looked fine, though it's using a custom title bar.

![Bildschirmfoto_2021-12-08_13-24-39](https://user-images.githubusercontent.com/23431373/145208202-cb55961b-cab8-4ccf-869d-13d685f793a3.png)

> also does change ui_scale in the settings something with that?

Changing ui_scale to 2.0 makes the UI bigger and slightly visible but still leaves everything obscured by the same amount. But now I can tell that the mouse also acts as if everything was where it' supposed to be.
![Bildschirmfoto_2021-12-08_15-21-12](https://user-images.githubusercontent.com/23431373/145224128-9d438be0-05cc-482e-a97b-46b00eacb04d.png)

**Comment by KottV** [comment]
https://github.com/helio-fm/helio-workstation 
https://github.com/sonosaurus/sonobus
https://github.com/mosynthkey/Melissa/

**Comment by OPNA2608** [comment]
I checked our helio-workstation package (v3.6, slightly outdated):

![Bildschirmfoto von 2021-12-09 23-17-45](https://user-images.githubusercontent.com/23431373/145484581-1504a220-9667-4c58-b2bf-8c6af3e735dc.png)

Which seems to looks like the example from their readme, no cut-off menu bar or anything:

![image](https://user-images.githubusercontent.com/23431373/145483667-6a9f5c99-299f-482d-88a1-a5865608e34f.png)

**Comment by follower** [comment]
FWIW I'm using the AppImage version of the v1.1.0 release on elementary OS v5.1 successfully.

From looking at the second image in https://github.com/BespokeSynth/BespokeSynth/issues/680#issuecomment-988858845 my guess is that something isn't accounting for the height of the title bar when calculating the Y-offset for the window content.

Based on my own recent experience with libxcb on Pantheon there does seem to be some slightly odd handling of offsets...

...oh, I just remembered, it might *also* be something not taking account of the height of the black/translucent "wingpanel"(?) at the top of the screen.

What does it look like in fullscreen mode?

**Comment by follower** [comment]
Oh. yeah, and just looking at my notes the Y-offset may change between maximised/non-maximised/full-screen window states.

**Comment by OPNA2608** [comment]
> What does it look like in fullscreen mode?

Not much different than the image in the OP. But just in case, here's a full desktop screenshot with the application in fullscreen mode (pressing the fullscreen button & pressing `/` looks identical). There doesn't seem to be a real, take-up-the-entire-screen kind of fullscreen mode unless I'm missing something?

![Bildschirmfoto von 2021-12-29 22-34-20](https://user-images.githubusercontent.com/23431373/147705089-94b241b1-b20b-4fd7-a96d-ac6ec41874d4.png)

**Comment by OPNA2608** [comment]
I checked on elementary OS 6.1 and couldn't reproduce so I debugged abit from there. Turns out this is caused by missing libraries in our packaging that the project doesn't explicitly link again & JUCE expects to be `dlopen`able at runtime. This seems to be a common "problem" with JUCE applications on our distro, but this is the first package where it has caused any issues in my testing.

Since this is an issue with JUCE's treatment of its X11 dependencies, I'll close this issue. This could be worked around in bespokesynth by explicitly linking against the libraries, but I don't know if JUCE has a good reason for this behaviour?

The following are the details of the problem, in case it helps anyone with their own debugging.

---

The ELF binary itself only depends on these X11 libraries.
![Bildschirmfoto von 2022-01-16 19-07-13](https://user-images.githubusercontent.com/23431373/149672227-9c91b37f-76de-473a-8aa2-f9705d8e4070.png)

But at runtime, JUCE tries to `dlopen` some more X11 libraries: libXcursor, libXinerama, libXrandr & libXScrnSaver.
![Bildschirmfoto von 2022-01-16 19-08-52](https://user-images.githubusercontent.com/23431373/149672274-f03d5095-a44d-41d5-a590-95d1d4b72737.png)

Due to how our packaging works, these libraries will normally only be made available to binaries & libraries when they explicitly depend on them like in the above picture. The workaround we'll go with on our end is to add a LD_LIBRARY_PATH wrapper to make the extra packages explicitly available.
